pkg/api/types.go

```diff
@@ -2051,7 +2051,7 @@ type SecurityContext struct {
 	// RunAsNonRoot indicates that the container should be run as a non-root user.  If the RunAsUser
 	// field is not explicitly set then the kubelet may check the image for a specified user or
 	// perform defaulting to specify a user.
-	RunAsNonRoot bool
+	RunAsNonRoot bool `json:"runAsNonRoot,omitempty"`
```

Comment:
Can you make this a pointer so that we can distinguish between unset and default value, and to be consistent with the unwritten rule that omitempty things should be pointers?

It needs a little more thought, but I think this is a safe-ish change:
  - serialized objects don't know what a pointer is.
  - Go client code that uses this field will break, but we don't make any versioning guarantees about Go client code.  
 - probably requires changes in openshift code.